src/Compilers/CSharp/Portable/Parser/LanguageParser_InterpolatedString.cs

```diff
@@ -39,98 +40,234 @@ private ExpressionSyntax ParseInterpolatedStringToken()
+                    ? node.WithDiagnosticsGreen(new[] { indentationError })
+                    : node;
             }
 
-            SyntaxToken getCloseQuote(Range openQuoteRange)
+            SyntaxToken getCloseQuote()
             {
                 // Make a token for the close quote " (even if it was missing)
-                var closeQuoteText = originalText[closeQuoteRange];
-                return closeQuoteText == ""
-                    ? SyntaxFactory.MissingToken(leading: null, SyntaxKind.InterpolatedStringEndToken, trailing: originalToken.GetTrailingTrivia())
-                    : SyntaxFactory.Token(leading: null, SyntaxKind.InterpolatedStringEndToken, closeQuoteText, closeQuoteText, originalToken.GetTrailingTrivia());
+                return TokenOrMissingToken(
+                    leading: null,
+                    kind switch
+                    {
+                        Lexer.InterpolatedStringKind.Normal => SyntaxKind.InterpolatedStringEndToken,
+                        Lexer.InterpolatedStringKind.Verbatim => SyntaxKind.InterpolatedStringEndToken,
+                        Lexer.InterpolatedStringKind.SingleLineRaw => SyntaxKind.InterpolatedSingleLineRawStringEndToken,
+                        Lexer.InterpolatedStringKind.MultiLineRaw => SyntaxKind.InterpolatedMultiLineRawStringEndToken,
+                        _ => throw ExceptionUtilities.UnexpectedValue(kind),
+                    },
+                    originalText[closeQuoteRange],
+                    originalToken.GetTrailingTrivia());
             }
         }
 
+        private static SyntaxToken TokenOrMissingToken(GreenNode? leading, SyntaxKind kind, string text, GreenNode? trailing)
+            => text == ""
+                ? SyntaxFactory.MissingToken(leading, kind, trailing)
+                : SyntaxFactory.Token(leading, kind, text, trailing);
+
+        private static int SkipWhitespace(ReadOnlySpan<char> text, int currentIndex)
+        {
+            while (currentIndex < text.Length && SyntaxFacts.IsWhitespace(text[currentIndex]))
+                currentIndex++;
+            return currentIndex;
+        }
+
+        private static int ConsumeRemainingContentOnLine(StringBuilder content, ReadOnlySpan<char> text, int currentIndex)
+        {
+            var start = currentIndex;
+            while (currentIndex < text.Length)
+            {
+                var ch = text[currentIndex];
+                if (!SyntaxFacts.IsNewLine(ch))
+                {
+                    currentIndex++;
+                    continue;
+                }
+
+                currentIndex += SlidingTextWindow.GetNewLineWidth(ch, currentIndex + 1 < text.Length ? text[currentIndex + 1] : '\0');
+                break;
+            }
+
+            var slice = text[start..currentIndex];
+#if NETCOREAPP
+            content.Append(slice);
+#else
+            unsafe
+            {
+                fixed (char* pointer = slice)
```

Comment:
Is this an established pattern for dealing with spans in older framework? 
I may be wrong but this feels dangerous. What if the GC moves the slice between this line and the next? Do we need to get a fixed reference first (with `GetPinnableReference`)?